### ISSUE TITLE
[RFE] unlock "pecl" vendor name

### ISSUE THREAD

**Issue by remicollet** [composer/packagist#1496]
Hi,

I started using the **pecl** vendor name, as I think it makes sense to use it for an extension existing in pecl and maintained by the PHP group.

I don't really like the idea of using vendor = extension, creating tons of vendors (ex: apcu/apcu)

Sadly, now the pecl vendor is locked

So I think it will be nice to unlock this vendor to allow other extension maintainers to add more.
Perhaps another way exists and may be better.

Open for discussion

**Comment by stof** [comment]
There's already many extensions registered using the `pecl` vendor name. So I don't think it has any special blockage.

The common rule for vendor names still apply: you cannot register a new package under an existing vendor name if you are not part of the Packagist maintainers of one of the existing packages of that namespace.

**Comment by remicollet** [comment]
> There's already several extensions registered using the pecl vendor name.

All from me :)

> So I don't think it has any special blockage.

And for `pecl/parallel` I had to create it, and give it to realFlowControl

> The common rule for vendor names still apply: you cannot register a new package under an existing vendor name if you are not part of the Packagist maintainers of one of the existing packages of that namespace.

Yes, this is the rule I propose to change, only for pecl.

**Comment by stof** [comment]
Note that removing this restriction on the vendor name would then mean that anyone can submit a `pecl/*` package, even for packages that are not part of the `php` organization on GitHub (and even packages that would not be PHP extensions at all). Is this really the expected behavior ?

**Comment by remicollet** [comment]
> Is this really the expected behavior ?

I don't know, perhaps some other restriction needed (ex sources under https://github.com/php)

Of course, after adding some ext (by me for others), we'll probably have enough allowed maintainers to add more.

**Comment by derickr** [comment]
I thought we had this discussion beforehand, and decided against using `pecl` as a vendor, I'm certain. Maybe @asgrim has the write up?

**Comment by derickr** [comment]
Ok, seems not (https://github.com/php/pie/issues/650.

I do think we should no longer pursue the name "pecl" though. And I think I do believe having each extension has its own vendor now makes more sense, as it would allow for every maintainer to do their own thing.

**Comment by asgrim** [comment]
I disagree with unlocking the `pecl/` vendor. The `pecl/` vendor should really be limited to exts in the PHP GitHub org, not just open for anyone.

As a rule of thumb, any extension should use their own vendor (e.g. Datadog could use `datadog/`, Xdebug uses `xdebug/`); bear in mind it is possible for a single vendor to have multiple packages. This is the same theory as Composer uses, so IMO nothing changes in the case of PIE.

The exception to this is those exts already hosted in the PHP GitHub organisation (i.e. https://github.com/php), e.g. yaml, mcrypt, timezonedb etc.. As per php/pie#65 - we had discussed these, and in that issue and both in the PHP Foundation discussion channel, the only consensus was that `php/` should not be used. Therefore I accepted it as `pecl/` instead.

Once some different maintainers have been added to allowed under `pecl/` vendor, I think this will be a non-issue.

**Comment by cmb69** [comment]
Given my concerns raised in https://github.com/ThePHPF/pie-design/issues/27, I would appreciate to use the `pecl` namespace only for extensions hosted on https://github.com/php.

**Comment by Seldaek** [comment]
Yup I tend to agree that opening this to anyone is bad as the PECL name carries some implicit trust in the eyes of many.

**Comment by sebastianbergmann** [comment]
> I do think we should no longer pursue the name "pecl" though.

:+1:

To me, what @asgrim, @cmb69, and @Seldaek wrote about, is about perception. Only extensions already hosted in the PHP GitHub organisation should be allowed to use `pecl` and we should strive for getting rid of the name there as well. Ultimately, I would hope for 0 extensions under the name `pecl`. Most importantly, though, no "random" and/or new extensions should be allowed to use that name.

**Comment by remicollet** [comment]
Sorry, but I think we need a _vendor_ for extensions maintained by the PHP community, and **pecl** makes sense.

I use "remi" for extensions I've created and I maintain. I don't want to use it for adopted extensions.

**Comment by cmb69** [comment]
> […] and we should strive for getting rid of the name there as well. Ultimately, I would hope for 0 extensions under the name `pecl`. Most importantly, though, no "random" and/or new extensions should be allowed to use that name.

How is this supposed to work in practice? There are [more than 80 extensions](https://github.com/orgs/php/repositories?q=pecl) under https://github.com/php. Are these supposed to be moved to some other namespace in the long run? Who would do this? Which namespace? And what if php-src unbundles further extensions? So far we moved them to a repo under github.com/php; if there was a maintainer, we would likely not have unbundled the extensions. And what about abandoned extensions living elsewhere (I think that @remicollet is concerned about these)? We could move them to github.com/php, but if we can't use the "pecl" namespace for those, we're back to square one.

**Comment by asgrim** [comment]
This came up yesterday on internals in fact:

https://externals.io/message/128536#128614

> I think the conclusion we came to was that existing PECL extensions may use
the pecl/ vendor, but new extns should use their own vendor. We can
probably just add that to the PIE docs for clarity, rather than use the RFC
process.

**Comment by Seldaek** [comment]
Ok so let's leave as is.. And if anyone needs help to submit an extension send them my way or submit for them and then add them as maintainer
